Guides/Transactions.md

```diff
@@ -0,0 +1,111 @@
+# Swift Driver Transactions Guide
+
+`MongoSwift` 1.0.0 added support for [transactions](https://docs.mongodb.com/manual/core/transactions/), which allow applications to use multi-statement transactions that guarantee the atomicity of reads and writes to multiple documents (in a single or multiple collections). Applications can use transactions instead of implementing complex and error-prone ACID-compliant logic themselves, simplifying development and allowing developers to focus on the new features that really matter.
+
+**Note**: Transactions only work with MongoDB replica sets (v4.0+) and sharded clusters (v4.2+).
+
+## Examples
+
+### Transaction that Atomically Moves a `Document` from One `MongoCollection` to Another
```

Comment:
each of these examples could use a little blurb describing what's going on in them. For this one, I think it might be useful to include an explanation of why a transaction is useful here, with perhaps a "bad" example of doing this non atomically with commentary pointing out the issues.

Reply:
Done.